File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 1/Python/web_table_test.py
from selenium import webdriver
from array import *
import time, unittest

class WebTableTest (unittest.TestCase):
	def setUp(self)	:
		self.driver = webdriver.Chrome()
		self.driver.get("http://dl.dropbox.com/u/55228056/Locators.html")
		
	def test_web_table(self):
		driver = self.driver
		
		simpleTable = driver.find_element_by_id("items" )
		
		#Get all rows
		rows = simpleTable.find_elements_by_tag_name("tr")
		self.assertEquals(3, len(rows))
		
		#Print data from each row
		for row in rows:
			cols = row.find_elements_by_tag_name("td")
			self.assertEquals(2, len(cols))
			line = []
			for col in cols:
				line.append(col.text)
			print "\t".join(line)
		
	def tearDown(self):
		self.driver.close()
		
if __name__ == "__main__":
	unittest.main()

File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 1/Ruby/web_table_test.rb
require 'rubygems'
require 'selenium-webdriver'
require 'test/unit'

class WebTableTest < Test::Unit::TestCase
	def setup
		@driver = Selenium::WebDriver.for :chrome
		@driver.get 'http://dl.dropbox.com/u/55228056/Locators.html'
		@verification_errors = []
	end
	
	def test_web_table
		simpleTable = @driver.find_element(:id => "items" )
		
		#Get all rows
		rows = simpleTable.find_elements(:tag_name => "tr")
		assert_equal 3, rows.length
		
		#Print data from each row
		for row in rows
			cols = row.find_elements(:tag_name => "td")
			assert_equal 2, cols.length
			line = []
			for col in cols
				line << col.text + "\t"
			end
			puts line.join
		end
	end
	
	def teardown
		@driver.close
		assert_equal [], @verification_errors
	end
end